waku/v2/node/waku_node.nim

```diff
@@ -371,47 +373,32 @@ proc startRelay*(node: WakuNode) {.async.} =
   info "relay started successfully"
 
 proc mountRelay*(node: WakuNode,
-                 topics: seq[string] = newSeq[string](),
+                 topics: seq[string] = @[],
                  triggerSelf = true,
-                 peerExchangeHandler = none(RoutingRecordsHandler))
-  # TODO: Better error handling: CatchableError is raised by `waitFor`
-  {.async, gcsafe, raises: [Defect, InitializationError, LPError, CatchableError].} =
-
-  proc msgIdProvider(m: messages.Message): Result[MessageID, ValidationResult] =
-    let mh = MultiHash.digest("sha2-256", m.data)
-    if mh.isOk():
-      return ok(mh[].data.buffer)
-    else:
-      return ok(($m.data.hash).toBytes())
-
-  let wakuRelay = WakuRelay.init(
-    switch = node.switch,
-    msgIdProvider = msgIdProvider,
-    triggerSelf = triggerSelf,
-    sign = false,
-    verifySignature = false,
-    maxMessageSize = MaxWakuMessageSize
+                 peerExchangeHandler = none(RoutingRecordsHandler)) {.async, gcsafe.} =
+  ## The default relay topics is the union of all configured topics plus default PubsubTopic(s)
+  info "mounting relay protocol"
+  
+  let initRes = WakuRelay.new(
+    node.peerManager,
+    defaultPubsubTopics = concat(@[DefaultPubsubTopic], topics),
+    triggerSelf = triggerSelf
   )
-  
-  info "mounting relay"
+  if initRes.isErr():
+    error "failed mountin relay protocol", error=initRes.error
+    return
 
-  ## The default relay topics is the union of
-  ## all configured topics plus the hard-coded defaultTopic(s)
-  wakuRelay.defaultPubsubTopics = concat(@[DefaultPubsubTopic], topics)
+  node.wakuRelay = initRes.value
 
   ## Add peer exchange handler
   if peerExchangeHandler.isSome():
-    wakuRelay.parameters.enablePX = true # Feature flag for peer exchange in nim-libp2p
-    wakuRelay.routingRecordsHandler.add(peerExchangeHandler.get())
+    node.wakuRelay.parameters.enablePX = true # Feature flag for peer exchange in nim-libp2p
+    node.wakuRelay.routingRecordsHandler.add(peerExchangeHandler.get())
 
-  node.wakuRelay = wakuRelay
   if node.started:
-    # Node has started already. Let's start relay too.
     await node.startRelay()
 
-  node.switch.mount(wakuRelay, protocolMatcher(WakuRelayCodec))    
-        
-  info "relay mounted successfully"
+  node.switch.mount(node.wakuRelay, protocolMatcher(WakuRelayCodec))
```

Comment:
I removed it to keep the same pattern as the other protocols. We are signaling the start of the mounting and the start of the protocol; additionally, if any of the steps fail, we log the error. So, in this case, I would say: "no news, good news".

But this is an excellent point: we need to review the logging in the node start sequence to make it consistent and see if we have any gaps. This is our first debugging tool. And this could be something to do before the next release.

Said that, shall I restore it? 😁

Reply:
I think so, if only for the fact that I just used that very log to confirm that relay wasn't mounted [while investigating this bug](https://github.com/status-im/nwaku/issues/1085) :D

Reply:
Restored 👍🏼